app/services/kredits.js

```diff
@@ -83,21 +87,23 @@ export default Service.extend({
 
   setup() {
     return this.getEthProvider().then((providerAndSigner) => {
-
       let kredits = new Kredits(providerAndSigner.ethProvider, providerAndSigner.ethSigner, {
         addresses: { Kernel: config.kreditsKernelAddress },
         apm: config.kreditsApmDomain,
         ipfsConfig: config.ipfs
       });
       return kredits
         .init()
-        .then((kredits) => {
+        .then(async (kredits) => {
           this.set('kredits', kredits);
+          this.set('currentBlock', await kredits.provider.getBlockNumber());
```

Comment:
do we actually need to await this here? 
currentBlock is not used in this function it seems?

Reply:
We don't want to set it to a promise I think.

Reply:
ok, I thought we want to set a promise that ember automatically resolves... like `getContributions` etc. return promises.

Reply:
That's different. `getContributions` doesn't set a property directly. Wherever it's used, the promise is actually handled.

Reply:
(That is to say the model hooks are promise-aware)